tonic/src/codec/compression.rs

```diff
@@ -10,35 +11,53 @@ pub(crate) const ACCEPT_ENCODING_HEADER: &str = "grpc-accept-encoding";
 /// Struct used to configure which encodings are enabled on a server or channel.
 #[derive(Debug, Default, Clone, Copy)]
 pub struct EnabledCompressionEncodings {
+    #[cfg(feature = "gzip")]
     pub(crate) gzip: bool,
 }
 
 impl EnabledCompressionEncodings {
-    /// Check if `gzip` compression is enabled.
-    pub fn gzip(self) -> bool {
-        self.gzip
+    /// Check if a [`CompressionEncoding`] is enabled.
+    pub fn is_enabled(&self, encoding: CompressionEncoding) -> bool {
+        match encoding {
+            #[cfg(feature = "gzip")]
+            CompressionEncoding::Gzip => self.gzip,
+        }
```

Comment:
how does this compile without the gzip feature?

Reply:
Without the `gzip` feature, `CompressionEncoding` becomes an enum with no variants, so it can't be constructed, similar to `core::convert::Infallible`. Thus, this match, even with no arms, is exhaustive. The Rust compiler assigns the `!` type to these types of `match` expressions which are exhaustive, but have no arms.

I use a similar strategy for `aliri`, which _can_ be build without support for any signing algorithm.